src/porepy/models/boundary_condition.py

```diff
@@ -0,0 +1,64 @@
+import porepy as pp
+
+from abc import ABC, abstractmethod
+import numpy as np
+
+
+from typing import Callable, Sequence
+
+
+class BoundaryConditionMixin(ABC):
+    """TODO"""
+
+    mdg: pp.MixedDimensionalGrid
+
+    @abstractmethod
+    def set_boundary_conditions(self, initial: bool) -> None:
+        """This method is called before a new time step to set the values of the
+        boundary conditions. The specific boundary condition values should be updated in
+        concrete overrides. By default, it does nothing.
+
+        Note:
+            One can use the convenience method `_update_boundary_condition` for each
+            boundary condition value.
+
+        """
+
+    def _update_boundary_condition(
```

Comment:
Still not convinced by name. Could we discuss in person, please?

Reply:
Following what we discussed in person, I got rid of the parameter `initial`, and renamed the method back to "update".

Reply:
To me, it seems the "initial" is still a parameter. Might I suggest that the docstring explains that this method "updates" in the same sense as for a dictionary? And that we make this method public, as it is something the user will be exposed to?

Reply:
I agree, I'll come back to adding the docstring later, so keeping this conversation open for now.

Reply:
Can we resolve, @Yuriyzabegaev?

Reply:
The docstrings are updated, so resolving it